source/server/hot_restart_impl.h

```diff
@@ -139,8 +139,9 @@ class HotRestartImpl : public HotRestart,
   static std::string hotRestartVersion(uint64_t max_num_stats, uint64_t max_stat_name_len);
 
   // RawStatDataAllocator
-  Stats::RawStatData* alloc(const std::string& name) override;
+  Stats::RawStatData* alloc(absl::string_view name) override;
   void free(Stats::RawStatData& data) override;
+  bool requiresBoundedStatNameSize() const override { return true; }
```

Comment:
Why is this overridden here and in `RawStatDataAllocator`?  Also why even have the `RawStatDataAllocator` intermediate class?

Reply:
Good catch - I didn't need this extra impl of requiresBoundedStatNameSize here, but I do need RawStatDataAllocator as it is used as a base class for test allocators as well as this.